Bottom: 19ac1839f9a10aa64938cf70ca15c369694ff1dc
Top:    863ec5158acd16395c5ff4282d579df6390093e1
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-07 15:00:57 -0500

Refresh of fixup-tmp-rework-ip-address-network

---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
new file mode 100644
index 00000000..77d9ce6a
--- /dev/null
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -0,0 +1,66 @@
+---
+
+svn_ip_addr: "{{ daedalus_subnet | ipaddr('101') | ipaddr('address') }}"
+svn_hostname: svn.novatech-llc.com
+svn_port_args:
+- "{{ svn_ip_addr }}:80:80"
+- "{{ svn_ip_addr }}:443:443"
+
+gitlab_ip_addr: "{{ daedalus_subnet | ipaddr('102') | ipaddr('address') }}"
+gitlab_hostname: git.novatech-llc.com
+gitlab_port_args:
+- "{{ gitlab_ip_addr }}:80:80"
+- "{{ gitlab_ip_addr }}:443:443"
+- "{{ gitlab_ip_addr }}:22:22"
+
+wiki_ip_addr: "{{ daedalus_subnet | ipaddr('103') | ipaddr('address') }}"
+wiki_hostname: wiki.novatech-llc.com
+wiki_port_args:
+- "{{ wiki_ip_addr }}:80:80"
+- "{{ wiki_ip_addr }}:443:443"
+
+mantisbt_ip_addr: "{{ daedalus_subnet | ipaddr('104') | ipaddr('address') }}"
+mantisbt_hostname: mantis.novatech-llc.com
+mantisbt_port_args:
+- "{{ mantisbt_ip_addr }}:80:80"
+- "{{ mantisbt_ip_addr }}:443:443"
+
+build_ip_addr: "{{ daedalus_subnet | ipaddr('105') | ipaddr('address') }}"
+build_hostname: buildsystem.novatech-llc.com
+build_port_args:
+- "{{ build_ip_addr }}:80:80"
+- "{{ build_ip_addr }}:443:443"
+
+testssh_ip_addr: "{{ daedalus_subnet | ipaddr('106') | ipaddr('address') }}"
+testssh_hostname: testssh.novatech-llc.com
+testssh_port_args:
+- "{{ testssh_ip_addr }}:2200:22"
+
+exim4_ip_addr: "{{ daedalus_subnet | ipaddr('107') | ipaddr('address') }}"
+exim4_hostname: mail.novatech-llc.com
+exim4_port_args:
+- "{{ exim4_ip_addr }}:25:25"
+- "{{ exim4_ip_addr }}:465:465"
+- "{{ exim4_ip_addr }}:587:587"
+
+openldap_ip_addr: "{{ daedalus_subnet | ipaddr('108') | ipaddr('address') }}"
+openldap_hostname: ldap.novatech-llc.com
+openldap_port_args:
+- "{{ openldap_ip_addr }}:389:389"
+- "{{ openldap_ip_addr }}:636:636"
+
+phpldapadmin_ip_addr: "{{ daedalus_subnet | ipaddr('108') | ipaddr('address') }}"
+phpldapadmin_hostname: ldap.novatech-llc.com
+phpldapadmin_port_args:
+- "{{ phpldapadmin_ip_addr }}:80:80"
+- "{{ phpldapadmin_ip_addr }}:443:443"
+
+phpmyadmin_ip_addr: "{{ daedalus_subnet | ipaddr('109') | ipaddr('address') }}"
+phpmyadmin_hostname: phpmyadmin.novatech-llc.com
+
+buildbot_ip_addr: "{{ daedalus_subnet | ipaddr('110') | ipaddr('address') }}"
+buildbot_hostname: buildbot.novatech-llc.com
+buildbot_port_args:
+- "{{ buildbot_ip_addr }}:80:8080"
+- "{{ buildbot_ip_addr }}:443:8443"
+- "{{ buildbot_ip_addr }}:9989:9989"
diff --git a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
index cf366ea4..f567d255 100644
--- a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
@@ -1,32 +1,4 @@
 ---
 
-svn_network:
-  ip_addr: 172.16.0.101
-  hostname: svn.novatech-llc.com
-gitlab_network:
-  ip_addr: 172.16.0.102
-  hostname: git.novatech-llc.com
-wiki_network:
-  ip_addr: 172.16.0.103
-  hostname: wiki.novatech-llc.com
-mantisbt_network:
-  ip_addr: 172.16.0.104
-  hostname: mantis.novatech-llc.com
-build_network:
-  ip_addr: 172.16.0.105
-  hostname: buildsystem.novatech-llc.com
-testssh_network:
-  ip_addr: 172.16.0.106
-  hostname: testssh.novatech-llc.com
-exim4_network:
-  ip_addr: 172.16.0.107
-  hostname: mail.novatech-llc.com
-openldap_network:
-  ip_addr: 172.16.0.108
-  hostname: ldap.novatech-llc.com
-phpldapadmin_network:
-  ip_addr: 172.16.0.108
-  hostname: ldap.novatech-llc.com
-phpmyadmin_network:
-  ip_addr: 172.16.0.109
-  hostname: phpmyadmin.novatech-llc.com
+daedalus_subnet: 172.16.0.0/16
+daedalus_net_connection: "Bridge br0"
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
index a12d209b..5111648b 100644
--- a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
@@ -1,69 +1,4 @@
 ---
 
-svn_network:
-  ip_addr: 172.16.0.201
-  hostname: svn.novatech-llc.com
-  port_args:
-    - '{{ svn_network.ip_addr }}:80:80'
-    - '{{ svn_network.ip_addr }}:443:443'
-
-gitlab_network:
-  ip_addr: 172.16.0.202
-  hostname: git.novatech-llc.com
-  port_args:
-    - '{{ gitlab_network.ip_addr }}:80:80'
-    - '{{ gitlab_network.ip_addr }}:443:443'
-    - '{{ gitlab_network.ip_addr }}:22:22'
-
-wiki_network:
-  ip_addr: 172.16.0.203
-  hostname: wiki.novatech-llc.com
-  port_args:
-    - '{{ wiki_network.ip_addr }}:80:80'
-    - '{{ wiki_network.ip_addr }}:443:443'
-
-mantisbt_network:
-  ip_addr: 172.16.0.204
-  hostname: mantis.novatech-llc.com
-  port_args:
-    - '{{ mantisbt_network.ip_addr }}:80:80'
-    - '{{ mantisbt_network.ip_addr }}:443:443'
-
-build_network:
-  ip_addr: 172.16.0.205
-  hostname: buildsystem.novatech-llc.com
-  port_args:
-    - '{{ build_network.ip_addr }}:80:80'
-    - '{{ build_network.ip_addr }}:443:443'
-
-testssh_network:
-  ip_addr: 172.16.0.206
-  hostname: testssh.novatech-llc.com
-  port_args:
-    - '{{ testssh_network.ip_addr }}:2200:22'
-
-exim4_network:
-  ip_addr: 172.16.0.207
-  hostname: mail.novatech-llc.com
-  port_args:
-    - '{{ exim4_network.ip_addr }}:25:25'
-    - '{{ exim4_network.ip_addr }}:465:465'
-    - '{{ exim4_network.ip_addr }}:587:587'
-
-openopenldap_network:
-  ip_addr: 172.16.0.208
-  hostname: ldap.novatech-llc.com
-  port_args:
-    - '{{ openopenldap_network.ip_addr }}:389:389'
-    - '{{ openopenldap_network.ip_addr }}:636:636'
-
-phpldapadmin_network:
-  ip_addr: 172.16.0.208
-  hostname: ldap.novatech-llc.com
-  port_args:
-    - '{{ phpldapadmin_network.ip_addr }}:80:80'
-    - '{{ phpldapadmin_network.ip_addr }}:443:443'
-
-phpmyadmin_network:
-  ip_addr: 172.16.0.209
-  hostname: phpmyadmin.novatech-llc.com
+daedalus_subnet: 172.16.103.0/16
+daedalus_net_connection: "enp5s0f0"
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 2ec6350c..a32df6d4 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -31,6 +31,18 @@
     - base_images
   - name: Configure Docker networking
     role: docker-networking
+    net_connection: '{{ daedalus_net_connection }}'
+    subnet: '{{ daedalus_subnet }}'
+    services:
+    - svn
+    - openldap
+    - exim4
+    - gitlab
+    - wiki
+    - mantisbt
+    - build
+    - testssh
+    - buildbot
     tags:
     - docker_networking
   - name: Build and start the SVN Docker container
diff --git a/roles/docker-buildsystem/defaults/main.yml b/roles/docker-buildsystem/defaults/main.yml
index e91d9afb..b4341aa0 100644
--- a/roles/docker-buildsystem/defaults/main.yml
+++ b/roles/docker-buildsystem/defaults/main.yml
@@ -1,19 +1,19 @@
 # Variables that are used by the role, but defined at the playbook level.
 # docker_projects_dir
-# build_network.ip_addr
-# testssh_network.ip_addr
+# build_ip_addr
+# testssh_ip_addr
 # buildsystem_ssh_prod_user_password
 # buildsystem_ssh_root_user_password
 # buildsystem_database_root_password
 
 buildsystem:
     port_args: 
-      - '{{ build_network.ip_addr }}:80:80'
-      - '{{ build_network.ip_addr }}:443:443'
+      - '80'
+      - '443'
 
 test_ssh:
     port_args:
-    - '{{ testssh_network.ip_addr }}:2200:22'
+    - '22'
 
 # restore directories to temporarly store data being restored into docker containers
 backup_buildsystem_docker_dir: '{{ docker_backup_dir }}/buildsystem'
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
index 6a0fce5b..8b0fdae3 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
@@ -25,7 +25,7 @@
     detach: True
     image: "{{ name_prefix }}teststation_database_server_image"
     name: "{{ name_prefix }}TestStation_database_server"
-    ports: "{{ build_network.ip_addr }}:3306:3306"
+    ports: "{{ build_ip_addr }}:3306:3306"
     state: "{{ container_start_state_nondata }}"
     restart_policy: always
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
index c87c74b9..4e53ef96 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
@@ -14,7 +14,7 @@
     image: "{{ name_prefix }}teststation_http_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
     name: "{{ name_prefix }}TestStation_http_server"
-    ports: "{{ build_network.ip_addr }}:80:80"
+    ports: "{{ build_ip_addr }}:80:80"
     state: "{{ container_start_state_nondata }}"
     restart_policy: always
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
index 26bd4e8d..d7d9d760 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
@@ -20,7 +20,7 @@
     image: "{{ name_prefix }}teststation_ssh_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
     name: "{{ name_prefix }}TestStation_ssh_server"
-    ports: "{{ build_network.ip_addr }}:22:22"
+    ports: "{{ build_ip_addr }}:22:22"
     restart_policy: always
     state: "{{ container_start_state_nondata }}"
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
index b5c65df2..6da5de30 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
@@ -31,6 +31,6 @@
     detach: True
     image: "{{ name_prefix }}teststation_ssh_server_manualtest_image"
     name: "{{ name_prefix }}TestStation_ssh_server_manualtest"
-    ports: "{{ testssh_network.ip_addr }}:22:22"
+    ports: "{{ testssh_ip_addr }}:22:22"
     state: "{{ container_start_state_nondata }}"
     restart_policy: always
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
index 7801d2be..e36626f0 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
@@ -13,7 +13,7 @@
     detach: True
     image: "{{ name_prefix }}teststation_tftp_image"
     name: "{{ name_prefix }}TestStation_tftp_server"
-    ports: "{{ build_network.ip_addr }}:69:69/udp"
+    ports: "{{ build_ip_addr }}:69:69/udp"
     state: "{{ container_start_state_nondata }}"
     restart_policy: always
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/main.yml b/roles/docker-buildsystem/tasks/main.yml
index 8754bec6..065f10b9 100644
--- a/roles/docker-buildsystem/tasks/main.yml
+++ b/roles/docker-buildsystem/tasks/main.yml
@@ -2,8 +2,8 @@
 - assert:
     that:
     - docker_projects_dir is defined
-    - build_network.ip_addr is defined
-    - testssh_network.ip_addr is defined
+    - build_ip_addr is defined
+    - testssh_ip_addr is defined
     - buildsystem_ssh_prod_user_password is defined
     - buildsystem_ssh_root_user_password is defined
     - buildsystem_database_root_password is defined
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index da75e132..699b23f7 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -6,13 +6,6 @@
     name: docker
     system: yes
 
-- name: restart docker service
-  service:
-    name: docker
-    state: restarted
-    enabled: yes
-
-- import_tasks: ip_addr.yml
 - import_tasks: packages.yml
 - import_tasks: sshd_settings.yml
 - import_tasks: mariadb_settings.yml
diff --git a/roles/docker-exim4/defaults/main.yml b/roles/docker-exim4/defaults/main.yml
index e099ddb9..e498de00 100644
--- a/roles/docker-exim4/defaults/main.yml
+++ b/roles/docker-exim4/defaults/main.yml
@@ -1,13 +1,12 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-exim4_network:
-  ip_addr: '127.0.0.1'
-  hostname: mail.example.com
-  port_args: 
-    - '{{ exim4_network.ip_addr }}:25:25'
-    - '{{ exim4_network.ip_addr }}:465:465'
-    - '{{ exim4_network.ip_addr }}:587:587'
+exim4_ip_addr: '127.0.0.1'
+exim4_hostname: mail.example.com
+exim4_port_args: 
+  - '25'
+  - '465'
+  - '587'
 
 # the repository containing Dockerfile to build the image
 exim4_image_repo: https://github.com/novatechweb/docker-exim4.git
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index ddd2a45e..2872b14d 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -105,10 +105,10 @@
 - name: start container
   docker_container:
     name: '{{ exim4_container_name }}'
-    hostname: '{{ exim4_network.hostname }}'
+    hostname: '{{ exim4_hostname }}'
     networks:
       - name: '{{ docker_network_frontend }}'
-    ports: '{{ exim4_network.port_args }}'
+    ports: '{{ exim4_port_args }}'
     image: '{{ exim4_image_name }}:{{ docker_image_tag }}'
     restart_policy: always
     state: started
diff --git a/roles/docker-mantisbt/defaults/main.yml b/roles/docker-mantisbt/defaults/main.yml
index a7ba2221..95730ece 100644
--- a/roles/docker-mantisbt/defaults/main.yml
+++ b/roles/docker-mantisbt/defaults/main.yml
@@ -1,12 +1,12 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-mantisbt_network:
-  ip_addr: '127.0.0.1'
-  hostname: mantis.example.com
-  port_args:
-    - '{{ mantisbt_network.ip_addr }}:80:80'
-    - '{{ mantisbt_network.ip_addr }}:443:443'
+network:
+mantisbt_ip_addr: '127.0.0.1'
+mantisbt_hostname: mantis.example.com
+mantisbt_port_args:
+  - '80'
+  - '443'
 
 # database usernames and passwords
 mantisbt_db_user: novatech
diff --git a/roles/docker-mediawiki/defaults/main.yml b/roles/docker-mediawiki/defaults/main.yml
index 1ad4ad9a..68feb554 100644
--- a/roles/docker-mediawiki/defaults/main.yml
+++ b/roles/docker-mediawiki/defaults/main.yml
@@ -1,12 +1,11 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-wiki_network:
-  ip_addr: '127.0.0.1'
-  hostname: wiki.example.com
-  port_args:
-    - '{{ wiki_network.ip_addr }}:80:80'
-    - '{{ wiki_network.ip_addr }}:443:443'
+wiki_ip_addr: '127.0.0.1'
+wiki_hostname: wiki.example.com
+wiki_port_args:
+  - '80'
+  - '443'
 
 # database usernames and passwords
 wiki_db_user: novatech
diff --git a/roles/docker-networking/tasks/hosts.yml b/roles/docker-networking/tasks/hosts.yml
new file mode 100644
index 00000000..f96b3d8b
--- /dev/null
+++ b/roles/docker-networking/tasks/hosts.yml
@@ -0,0 +1,14 @@
+---
+# file: roles/docker-common/tasks/ip_addr.yaml
+# Add each service to the /etc/hosts file
+
+- name: "Build /etc/hosts file"
+  lineinfile: 
+    path: /etc/hosts
+    regexp: '.*{{ hostname }}$'
+    line:  '{{ ip_addr }} {{ hostname }}'
+    state: present
+  vars:
+    ip_addr: '{{ lookup("vars", item + "_ip_addr") }}'
+    hostname: '{{ lookup("vars", item + "_hostname") }}'
+  loop: "{{ services }}"
diff --git a/roles/docker-common/tasks/ip_addr.yml b/roles/docker-networking/tasks/ip_addr.yml
similarity index 80%
rename from roles/docker-common/tasks/ip_addr.yml
rename to roles/docker-networking/tasks/ip_addr.yml
index 5a98cc37..f5d4ff8b 100644
--- a/roles/docker-common/tasks/ip_addr.yml
+++ b/roles/docker-networking/tasks/ip_addr.yml
@@ -20,9 +20,14 @@
   register: network_pre
 
 - name: Add service IP addresses
-  command: nmcli connection modify '{{ network_connection }}' +ipv4.addresses '{{ item.value.ip_addr }}/16'
-  with_dict: "{{ container_addr_map }}"
-    
+  command: >
+    nmcli connection 
+    modify '{{ network_connection }}' 
+    +ipv4.addresses '{{ ip_addr }}/{{ subnet | ipaddr("prefix") }}'
+  vars:
+    ip_addr: '{{ lookup("vars", item + "_ip_addr") }}'
+  loop: "{{ services }}"
+ 
 - sysctl:
     name: net.ipv4.ip_forward
     value: 1
diff --git a/roles/docker-networking/tasks/main.yml b/roles/docker-networking/tasks/main.yml
index 595e8bb7..c220b585 100644
--- a/roles/docker-networking/tasks/main.yml
+++ b/roles/docker-networking/tasks/main.yml
@@ -1,6 +1,16 @@
 ---
 # tasks file for docker-networking
 
+- assert:
+    that:
+    - network_connection is defined
+    - subnet is defined
+    - services is defined
+
+- import_tasks: ip_addr.yml
+
+- import_tasks: hosts.yml
+
 - name: Create frontend network for services
   docker_network:
     name: '{{ docker_network_frontend }}'
diff --git a/roles/docker-openldap/defaults/main.yml b/roles/docker-openldap/defaults/main.yml
index f946155d..9e2fddc4 100644
--- a/roles/docker-openldap/defaults/main.yml
+++ b/roles/docker-openldap/defaults/main.yml
@@ -1,19 +1,17 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-openldap_network:
-  ip_addr: '127.0.0.1'
-  hostname: ldap.example.com
-  port_args:
-    - '{{ openldap_network.ip_addr }}:389:389'
-    - '{{ openldap_network.ip_addr }}:636:636'
-
-phpldapadmin_network:
-  ip_addr: '127.0.0.1'
-  hostname: ldap.example.com
-  port_args: 
-    - '{{ phpldapadmin_network.ip_addr }}:80:80'
-    - '{{ phpldapadmin_network.ip_addr }}:443:443'
+openldap_ip_addr: '127.0.0.1'
+openldap_hostname: ldap.example.com
+openldap_port_args:
+  - '389'
+  - '636'
+
+phpldapadmin_ip_addr: '127.0.0.1'
+phpldapadmin_hostname: ldap.example.com
+phpldapadmin_port_args: 
+  - '80'
+  - '443'
 
 openldap_backup_files:
   - ldif_file: LDAP_database_0000.ldif
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 4da8f5e6..cc7fe095 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -111,14 +111,14 @@
     name: '{{ openldap_container_name }}'
     detach: true
     restart_policy: always
-    hostname: '{{ openldap_network.hostname }}'
-    ports: '{{ ldap_network.port_args }}'
+    hostname: '{{ openldap_hostname }}'
+    ports: '{{ openldap_port_args }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ openldap_cv_name }}:/etc/ldap:z'
       - '{{ openldap_dv_name }}:/var/lib/ldap:z'
     env:
-      LDAP_HOSTNAME: '{{ openldap_network.hostname }}'
+      LDAP_HOSTNAME: '{{ openldap_hostname }}'
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
     networks:
       - name: '{{ docker_network_frontend }}'
@@ -128,14 +128,14 @@
     name: '{{ phpldapadmin_container_name }}'
     detach: true
     restart_policy: always
-    hostname: '{{ phpldapadmin_network.hostname }}'
-    ports: '{{ phpldapadmin_network.port_args }}'
+    hostname: '{{ phpldapadmin_hostname }}'
+    ports: '{{ phpldapadmin_port_args }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
     env:
       PHPLDAPADMIN_HTTPS: 'false'
       PHPLDAPADMIN_LDAP_HOSTS: 'ldap'
-      LDAP_HOSTNAME: '{{ container_addr_map.phpldapadmin.hostname }}'
+      LDAP_HOSTNAME: '{{ openldap_hostname }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     image: '{{ phpldapadmin_image_name }}:{{ docker_image_tag }}'
diff --git a/roles/docker-svn/defaults/main.yml b/roles/docker-svn/defaults/main.yml
index b4aba59c..9529f078 100644
--- a/roles/docker-svn/defaults/main.yml
+++ b/roles/docker-svn/defaults/main.yml
@@ -1,12 +1,11 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-svn_network:
-  ip_addr: '127.0.0.1'
-  hostname: svn.example.com
-  port_args: 
-    - '{{ svn_network.ip_addr }}:80:80'
-    - '{{ svn_network.ip_addr }}:443:443'
+svn_ip_addr: '127.0.0.1'
+svn_hostname: svn.example.com
+svn_port_args: 
+  - '80'
+  - '443'
 
 # The repositories that will be created/imported durring a data-volume restore
 svn_repos: 'example'
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index d1eeb954..183bb213 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -80,16 +80,16 @@
     name: '{{ svn_container_name }}'
     detach: true
     restart_policy: always
-    hostname: '{{ svn_network.hostname }}'
-    ports: '{{ svn_network.port_args }}'
+    hostname: '{{ svn_hostname }}'
+    ports: '{{ svn_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ svn_dv_name }}:/var/lib/svn:z'
     env:
-      SVN_HOSTNAME: '{{ svn_network.hostname }}'
-      SVN_LDAP_PASSWORD: '{{ ldap_proxyagent_password }}'
+      SVN_HOSTNAME: '{{ svn_hostname }}'
+      SVN_LDAP_PASSWORD: '{{ openldap_proxyagent_password }}'
     image: '{{ svn_image_name }}:{{ docker_image_tag }}'
 
 # *****************************************************************************
